test/functional/rpc_generateblock.py

```diff
@@ -6,26 +6,29 @@
 '''
 
 from test_framework.test_framework import BitcoinTestFramework
+from test_framework.address import ADDRESS_BCRT1_P2WSH_OP_TRUE
 from test_framework.util import (
     assert_equal,
     assert_raises_rpc_error,
 )
+from test_framework.wallet import MiniWallet
 
 
 class GenerateBlockTest(BitcoinTestFramework):
     def set_test_params(self):
         self.num_nodes = 1
 
-    def skip_test_if_missing_module(self):
-        self.skip_if_no_wallet()
-
     def run_test(self):
         node = self.nodes[0]
+        miniwallet = MiniWallet(node)
+        address = ADDRESS_BCRT1_P2WSH_OP_TRUE
 
         self.log.info('Generate an empty block to address')
-        address = node.getnewaddress()
         hash = node.generateblock(output=address, transactions=[])['hash']
         block = node.getblock(blockhash=hash, verbose=2)
+        tx = block['tx'][0]
+        utxo = {'txid': tx['txid'], 'vout': 0, 'value': tx['vout'][0]['value']}
+        miniwallet._utxos.append(utxo)
```

Comment:
Is this necessary? I don't see why we need this Coinbase in our utxos. I don't see us specifically spending it, and if you remove this it gets rid of the potential problem of appending to mini wallets private `_utxos`

Reply:
10 blocks isn't enough for the test because the 10 generated by MiniWallet are sent to sit in the mempool not to be mined. So I can up the generated blocks to 11 and remove that code, or do this approach.

Reply:
Is there any reason you need this specific utxo for this test? If you need 11, I'd say generate 11 blocks where you need them.

but the 10 generated by MiniWallet are definitely mined. ie if you synced another node, you'd see those blocks with the coinbases (utxos) you are spending. they aren't just in the mempool

if I'm missing something please lmk, definitely possible

Reply:
I do not need that specific utxo for the test. I guess I should have elaborated more on my point. What I meant was that those 10 blocks are generated and are sent via `send_self_transfer` however those transactions are never confirmed. So while those new utxos are present within miniwallet generating a confirmed transaction based on them (because they are unconfirmed) isn't possible. Please correct me if I am wrong. Its probably just a better approach to generate 11.

Reply:
Ah yeah, so the transactions you send with `send_self_transfer` are in the mempool. However, you could still spend them even though they are only in the mempool. And you could also generate a block, and all the transactions in the mempool would then be mined/confirmed (although I don't think this is necessary).

I'd recommend generating 11 blocks for miniwallet if you need to send 11 utxos. Otherwise the 11th will pick one of the previous `send_self_transfer` utxos to spend (ie one of the transactions in the mempool is the parent of that new "child" txn)